packages/grafana-ui/src/components/uPlot/types.ts

```diff
@@ -17,12 +20,11 @@ export interface PlotPluginProps {
 }
 
 export interface PlotProps {
-  data: DataFrame;
-  timeRange: TimeRange;
-  timeZone: TimeZone;
+  data: AlignedData;
   width: number;
   height: number;
   config: UPlotConfigBuilder;
```

Comment:
Does this need config builder?  could it take the config directly?

Reply:
It could, yes. Leaving it to a separate PR maybe not to grow this already big PR